### PR TITLE
Add unit tests for export.bin.checksum

### LINKED ISSUE #606: CWA Hash should also incorporate the signing information
We are currently using the `export.bin` payload in order to compute the checksum. This approach does not reflect the signing information, which is part of the `export.sig`. Therefore, if the key would be rolled at some point in time, the cwa hash will not change, and hence the system will not re-distribute the files with the new signature information.

Therefore, the key cwa hash should also reflect the key, which was used for signing they `export.bin` payload.

Current is:
https://github.com/corona-warn-app/cwa-server/blob/cd4eb1af6f8f60ae2b97759cbdeaaead84a20f13/services/distribution/src/main/java/app/coronawarn/server/services/distribution/assembly/structure/file/FileOnDiskWithChecksum.java#L75-L80

### PR DESCRIPTION
**Pull request by mibrasap**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn install` runs for the whole project and, if you touched any code in the respective service, submission and distribution service can be run with `spring-boot:run`

## Description

Add unit tests for export.bin.checksum. Checking if checksum is calculated deterministically and incorporates changes on the signature's verification key version.

closes #606

### CODE REVIEW

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  void testChecksumChangesWhenPrivateKeyIsModified() throws IOException {
+    File outputFile1 = createFile();
+    byte[] checksum1 = readChecksumFile(outputFile1);
+
+    modifySignaturePrivateKeyVersion();
+
+    File outputFile2 = createFile();
+    byte[] checksum2 = readChecksumFile(outputFile2);
+
+    assertThat(!(Arrays.equals(checksum1, checksum2)));
```
Comment (michael-burwig) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java:91]: Better use of assertions:
```suggestion
    assertThat(checksum1).isNotEqualTo(checksum2);
```

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+  @Test
+  void testChecksumIsDeterministic() throws IOException {
+    File outputFile1 = createFile();
+    byte[] checksum1 = readChecksumFile(outputFile1);
+
+    File outputFile2 = createFile();
+    byte[] checksum2 = readChecksumFile(outputFile2);
+
+    assertThat(Arrays.equals(checksum1, checksum2));
```
Comment (michael-burwig) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java]:
```suggestion
    assertThat(checksum1).isEqualTo(checksum2);
```

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+import org.junit.rules.TemporaryFolder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
```
Comment (michael-burwig) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java:48]: Consider moving before non-static imports for consistency

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+    File outputFile2 = createFile();
+    byte[] checksum2 = readChecksumFile(outputFile2);
+
+    assertThat(!(Arrays.equals(checksum1, checksum2)));
+  }
+
+  private TemporaryExposureKeyExportFile createTemporaryExposureKeyExportFile() {
+    return TemporaryExposureKeyExportFile.fromDiagnosisKeys(
+        Set.of(
+            DiagnosisKey.builder()
```
Comment (michael-burwig) [inline services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java:97]: Consider re-use of ``Helpers.buildDiagnosisKeyForSubmissionTimestamp`` or similar

File: services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java (hunk 1 above)
Comment (mibrasap) [reply services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java]: Checksum1 and checksum2 are arrays; `isEqualTo` and `isNotEqualTo` is checking for referential equality, which we do not want. We need a check for equal content of the arrays' content.

Comment (michael-burwig) [reply services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java]: I assume that ``isEqualTo`` is employing ``org.assertj.core.internal.ObjectArrayElementComparisonStrategy`` here, which performs a _propper_ array comparison.
That is also what I could observed when trying it.

Comment (michael-burwig) [reply services/distribution/src/test/java/app/coronawarn/server/services/distribution/assembly/diagnosiskeys/structure/file/TemporaryExposureKeyExportFileTest.java]: Okay, actually it uses ``Arrays.deepEquals`` internally.

**Pull request merged**